src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -48,6 +48,11 @@ Onyx.connect({
         }
 
         currentAccountID = val.accountID;
+        if (Navigation.isActiveRoute(ROUTES.SIGN_IN_MODAL)) {
+            // This means sign in in RHP was successful, so we can dismiss the modal and subscribe to user events
+            Navigation.dismissModal();
```

Comment:
Seems we are closing the modal early which causes the issue https://github.com/Expensify/App/issues/36976